src/open_vr/openvr_data.cpp

```diff
@@ -264,6 +395,406 @@ void openvr_data::get_eye_to_head_transform(godot_transform *p_transform, int p_
+		return false;
+	} else if (custom_actions[p_action_idx].handle == vr::k_ulInvalidActionHandle) {
+		// action doesn't exist
+		// printf("No handle set for action: %s\n", custom_actions[p_action_idx].name);
+		return false;
+	}
+
+	// let's retrieve it
+	vr::EVRInputError err = vr::VRInput()->GetPoseActionDataForNextFrame(custom_actions[p_action_idx].handle, vr::TrackingUniverseStanding, p_data, sizeof(vr::InputPoseActionData_t), vr::k_ulInvalidInputValueHandle);
+	if (err != vr::VRInputError_None) {
+		// printf("Couldn't retrieve pose %i\n", err);
+		return false;
+	}
+
+	return true;
+}
+
+bool openvr_data::get_custom_digital_data(int p_action_idx) {
+	if (p_action_idx < 0) {
+		// we never registered our handle
+		// printf("Index not setup: %i\n", p_action_idx);
+		return false;
+	} else if (p_action_idx >= custom_actions.size()) {
+		// index out of bounds
+		// printf("Index out of bounds: %i\n", p_action_idx);
+		return false;
+	} else if (custom_actions[p_action_idx].handle == vr::k_ulInvalidActionHandle) {
+		// action doesn't exist
+		// printf("No handle set for action: %s\n", custom_actions[p_action_idx].name);
+		return false;
+	}
+
+	// let's retrieve it
+	vr::InputDigitalActionData_t digital_data;
+	vr::EVRInputError err = vr::VRInput()->GetDigitalActionData(custom_actions[p_action_idx].handle, &digital_data, sizeof(digital_data), vr::k_ulInvalidInputValueHandle);
+	if (err != vr::VRInputError_None) {
+		// printf("Couldn't retrieve digital %i\n", err);
+		return false;
+	}
+
+	return digital_data.bActive && digital_data.bState;
+}
+
+godot::Vector2 openvr_data::get_custom_analog_data(int p_action_idx) {
+	if (p_action_idx < 0) {
+		// we never registered our handle
+		// printf("Index not setup: %i\n", p_action_idx);
+		return godot::Vector2();
+	} else if (p_action_idx >= custom_actions.size()) {
+		// index out of bounds
+		// printf("Index out of bounds: %i\n", p_action_idx);
+		return godot::Vector2();
+	} else if (custom_actions[p_action_idx].handle == vr::k_ulInvalidActionHandle) {
+		// action doesn't exist
+		// printf("No handle set for action: %s\n", custom_actions[p_action_idx].name);
+		return godot::Vector2();
+	}
+
+	// let's retrieve it
+	vr::InputAnalogActionData_t analog_data;
+	vr::EVRInputError err = vr::VRInput()->GetAnalogActionData(custom_actions[p_action_idx].handle, &analog_data, sizeof(analog_data), vr::k_ulInvalidInputValueHandle);
+	if (err != vr::VRInputError_None) {
+		// printf("Couldn't retrieve analog %i\n", err);
+		return godot::Vector2();
+	} else if (!analog_data.bActive) {
+		return godot::Vector2();
+	}
+
+	return godot::Vector2(analog_data.x, analog_data.y);
+}
+
+////////////////////////////////////////////////////////////////
+// Bind our default actions, if the actions file has been edit
+// and the defaults removed they will simply be undefined
+void openvr_data::bind_default_action_handles(tracked_device *p_device, godot_int p_hand) {
+	char *hand = p_hand == 1 ? "left" : "right";
```

Comment:
Got the following compile error here:

```
src/open_vr/openvr_data.cpp: In member function 'void openvr_data::bind_default_action_handles(openvr_data::tracked_device*, godot_int)':
src/open_vr/openvr_data.cpp:561:27: error: invalid conversion from 'const char*' to 'char*' [-fpermissive]
  char *hand = p_hand == 1 ? "left" : "right";
               ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
```
I solved that by writing const char *hand instead.